config/service_test.go

```diff
@@ -271,3 +271,124 @@ func TestServiceConfig_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestServiceConfigs_CTSUserDefinedMeta(t *testing.T) {
```

Comment:
These test cases were helpful to understand CTSUserDefinedMeta! I was trying to make sure I understood it and tested out:
```
{
	"service list value matches name or id",
	&ServiceConfigs{{
		Name: String("a"),
		CTSUserDefinedMeta: map[string]string{
			"a": "b",
		},
	}, {
		Name: String("a-with-meta"),
		ID:   String("a"),
		CTSUserDefinedMeta: map[string]string{
			"key": "value",
		},
	}},
	[]string{"a"},
	map[string]map[string]string{
		"a":  {"a": "b"},
	},
}
```
And got the result:
```
expected: map[string]map[string]string{"a":map[string]string{"a":"b"}}
actual  : map[string]map[string]string{"a":map[string]string{"a":"b"}, "a-with-meta":map[string]string{"key":"value"}}
```

It seems like this isn't expected. I was wondering if this test case could actually happen? Let me know if I'm misunderstanding!

Reply:
Good edge case! The result is not expected; however, `ServiceConfigs.Validate()` should catch that and error before proceeding to `ServiceConfigs.CTSUserDefinedMeta()`. This case would be considered an invalid set of services because there's a conflict in uniqueness between a service ID and service name.

I added another set of tests to make it a bit more clear, thanks for pointing this out!

Reply:
Ah, I see! Thanks for the tests and context on how this case is invalid.